sbin/spark-daemon.sh

```diff
@@ -162,6 +162,16 @@ run_command() {
   esac
 
   echo "$newpid" > "$pid"
+  
+  #here add this loop to check the daemon process turn into java process. It takes at most 5 seconds.
```

Comment:
This would be somewhat clearer as something like "Poll for up to 5 seconds for the java process to start" Otherwise LGTM